helpers/GenerateResourcesAndImage.ps1

```diff
@@ -266,16 +266,23 @@ Function GenerateResourcesAndImage {
                 $ResourceGroupExists = $false
             }
             else {
-                # Resource group already exists, ask the user what to do
-                $title = "Resource group '$ResourceGroupName' already exists"
-                $message = "Do you want to delete the resource group and all resources in it?"
+                # are we running in a non-interactive session?
+                # https://stackoverflow.com/questions/9738535/powershell-test-for-noninteractive-mode
+                if ([System.Console]::IsOutputRedirected -or ![Environment]::UserInteractive -or !!([Environment]::GetCommandLineArgs() | Where-Object { $_ -ilike '-noni*' })) {
```

Comment:
Do we need double `-not` here -  (**!!**([Environment]::GetCommandLineArgs() | Where-Object { $_ -ilike '-noni*' })))?

Reply:
can you elaborate on some example when it comes to

Reply:
I got it. It turns the value from expression to the boolean (True or False)